src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java

```diff
@@ -0,0 +1,172 @@
+package com.wikia.webdriver.pageobjectsfactory.componentobject;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.WikiaWebDriver;
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.networktrafficinterceptor.NetworkTrafficInterceptor;
+import com.wikia.webdriver.common.core.url.Page;
+import com.wikia.webdriver.common.core.url.UrlBuilder;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+
+import org.openqa.selenium.Cookie;
+import org.openqa.selenium.TimeoutException;
+import net.lightbody.bmp.core.har.HarEntry;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.time.Duration;
+import java.util.List;
+
+public class TrackingOptInModal extends BasePageObject {
+
+    @FindBy(css = "body > div:nth-child(5) > div > div")
+    WebElement modal;
+
+    @FindBy(css = "div._36mdTyonPU0bxbU70dDO4f")
+    WebElement acceptButton;
+
+    @FindBy(css = "div._3mGMvAUi8MtSBaB1BUvLtz")
+    WebElement rejecttButton;
+
+    @FindBy(css="div[data-tracking-opt-in-overlay]")
+    private WebElement optInModalOverlay;
+
+    private HarEntry entry;
+    private static final Duration WAITNG_TIME_FOR_ALL_REQUESTS = Duration.ofSeconds(10);
+    private static final String MODAL_INSTANT_GLOBAL = "InstantGlobals.wgEnableTrackingOptInModal=1";
+
+
+    public static void setGeoCookie(WikiaWebDriver driver, String continent, String country) {
+        Cookie geoCookie = driver.manage().getCookieNamed("Geo");
+        driver.manage().deleteCookie(geoCookie);
+        driver.manage().addCookie(new Cookie(
+            "Geo",
+            "{%22region%22:%22WP%22%2C%22country%22:%22" + country + "%22%2C%22continent%22:%22" + continent + "%22}",
+            String.format(".%s", Configuration.getEnvType().getWikiaDomain()),
+            null,
+            null
+        ));
+    }
+
+    public boolean isVisible() {
+        try {
+            wait.forElementVisible(optInModalOverlay);
+            PageObjectLogging.logInfo("Tracking modal visible");
+
+            return true;
+        } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)